### PR TITLE
[Work in Progress] Cleanse Handler

### PR DESCRIPTION
**Pull request by matteosuppo**
A new handler that allows to delete old and useless tokens.

It's not finished by I needed to know if I'm going in the right direction.

### CODE REVIEW

File: oauth2/fosite_store_memory.go [hunk 1]
```diff
@@ -182,3 +184,17 @@ func (s *FositeMemoryStore) RevokeAccessToken(ctx context.Context, id string) er
 	}
 	return nil
 }
+
+func (s *FositeMemoryStore) CleanseTokens(ctx context.Context, before time.Time) error {
+	errs := multierror.Accumulator{}
+
+	for sig, token := range s.AccessTokens {
+		if token.GetRequestedAt().Before(before) {
+			errs.Push(s.DeleteAccessTokenSession(ctx, sig))
```
Comment (arekkas) [inline oauth2/fosite_store_memory.go]: Is the idea here that you don't want to exit on error but rather process everything and then return the first error?

Comment (matteosuppo) [inline oauth2/fosite_store_memory.go]: Return all the errors. The accumulator returns an error containing a list of errors

**Comment by arekkas** [comment]
you need to add the new dependency using `glide update`

**Comment by arekkas** [comment]
If you ever come back to this issue feel free to comment & reopen

**Pull request closed without merging**